pkg/logql/blocker.go

```diff
@@ -33,7 +33,7 @@ func (qb *queryBlocker) isBlocked(ctx context.Context, tenant string) bool {
 		return false
 	}
 
-	query := qb.q.params.Query()
+	query := qb.q.params.QueryString()
```

Comment:
This actually opens up a nice possibility for doing AST-based blocking; for example blocking any `quantile_over_time` function without a grouping with absolute precision.

No idea how we'd configure that, but it's a nice consequence of this work.

Reply:
Nice catch! I refactored the `QueryType` method a little. The visitor would be ideal to walk the query. As for configuration I do not know. May JSON path for the AST would work.